app/src/main/java/fr/free/nrw/commons/upload/SpinnerLanguagesAdapter.kt

```diff
@@ -0,0 +1,105 @@
+package fr.free.nrw.commons.upload
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ArrayAdapter
+import androidx.annotation.LayoutRes
+import androidx.core.os.ConfigurationCompat
+import fr.free.nrw.commons.R
+import fr.free.nrw.commons.utils.BiMap
+import fr.free.nrw.commons.utils.LangCodeUtils
+import kotlinx.android.extensions.LayoutContainer
+import kotlinx.android.synthetic.main.row_item_languages_spinner.*
+import org.apache.commons.lang3.StringUtils
+import java.util.*
+
```

Comment:
Could we get Javadocs for new classes please? They should definitely have been in the original to begin with, but we have to start somewhere - if we are rewriting an entire class I think it is a good time to start including Javadocs as well.

Reply:
Is this satisfactory? I would never have written javadocs for any sustained length of time as every company I have worked in loathed them

Reply:
Yes, thank you! :)